s3stream/src/main/java/com/automq/stream/s3/wal/impl/block/BlockWALService.java

```diff
@@ -115,7 +116,11 @@ public class BlockWALService implements WriteAheadLog {
     public static final int WAL_HEADER_CAPACITY = WALUtil.BLOCK_SIZE;
     public static final int WAL_HEADER_TOTAL_CAPACITY = WAL_HEADER_CAPACITY * WAL_HEADER_COUNT;
     private static final Logger LOGGER = LoggerFactory.getLogger(BlockWALService.class);
-    private final AtomicBoolean started = new AtomicBoolean(false);
+    private static final int WAL_STATE_INIT = 1;
+    private static final int WAL_STATE_STARTED = 2;
+    private static final int WAL_STATE_SHUTING_DOWN = 3;
+    private static final int WAL_STATE_SHUTDOWN = 4;
+    private static final AtomicIntegerFieldUpdater<BlockWALService> WAL_STATE = AtomicIntegerFieldUpdater.newUpdater(BlockWALService.class, "state");
```

Comment:
I think there is no need to use `AtomicIntegerFieldUpdater` here; an `AtomicReference<State>` should suffice. Here are a few reasons:

- In our scenario, there won't be many `WAL` instances simultaneously (in fact, in most cases, there is only one). So a global `static` updater is unnecessary.
- Compared to `AtomicIntegerFieldUpdater`, `AtomicReference` can provide better code readability and maintainability.
- I conducted a simple benchmark, and there is no significant performance difference between them (in fact, for the most frequently called `get` method in our use case, `AtomicReference` performs slightly better).